route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -84,23 +85,64 @@ namespace route_following_plugin
             ROS_WARN_STREAM("Cannot find current lanelet in shortest path!");
             return true;
         }
+        
+       if(req.prior_plan.maneuvers.size()!=0){
+            current_time=req.prior_plan.planning_completion_time;
+            double dist_covered=0.0;
+            // find current progress, current_lanelet and total maneuver length
+            for(auto i=0;i<req.prior_plan.maneuvers.size();i++){
+               auto type= req.prior_plan.maneuvers[i].type;
+               cav_msgs::Maneuver maneuver_msg=req.prior_plan.maneuvers[i];
+                switch(type)
+                {
+                    case 0:
```

Comment:
I'd recommend using the enum value names rather than integer literals `cav_msgs::Maneuver::LANE_FOLLOWING` rather than `0` for readability